content/guides/ci-cd/cnbp-knative-argocd.md

```diff
@@ -0,0 +1,263 @@
+
+![Building and Running Applications on Kubernetes](../../../static/images/guides/ci-cd/cnbp-knative-argocd/dev-process.png)
+
+
+In this Guide we are going to show how to automate building of Container Images and buildout of Kubernetes Resources using Cloud Native BuildPacks and Knative respectively. We will also demonstrate how this can process can be further optimized using **Gitops** style practices via ArgoCD. We will work with a sample application based on Spring and run the application on Kubernetes with an addressable URL. We will use:
+
+- [Cloud Native Buildpacks](https://buildpacks.io/) to automate Container Image build process. Cloud Native Buildpacks is a specification that defines how OCI compliant containers can be build, removing the need to specify or build `Dockerfiles`. They can automatically detect the language an application is written in and determine the best and most secure way to package applications in a container image. Cloud Native Buildpacks can also be used to update container images easily for any changes. For this guide we will use an implementation of Cloud Native BuildPacks called [kpack](https://github.com/pivotal/kpack). Kpack lets you use deploy CLoud Native BuildPacks on a Kubernetes Cluster.
+
+- [Knative](https://knative.dev/) to automatically generate an Ingress Service with URL and other Kubernetes Resources for the container image that was built using Cloud Native Buildpacks. Kantive-serving automates the process of creating Kubernetes objects needed for an application like Deployment, Replicasets, Services etc. Eliminating the need to write complex Kubernetes YAML files. 
+
+- [ArgoCD](https://argoproj.github.io/) to automates deployment pipeline pushing container images on to Kubernetes using Knative. ArgoCD helps deploy application continuously using Gitops methodology. It can take specifications like Kubernetes resources, Knative, Kustomize etc. to deploy application on Kubernetes.
```

Comment:
```suggestion
- [ArgoCD](https://argoproj.github.io/) to automate deployment pipeline pushing container images on to Kubernetes using Knative. ArgoCD helps deploy application continuously using Gitops methodology. It can take specifications like Kubernetes resources, Knative, Kustomize etc. to deploy application on Kubernetes.
```